File: python/exercise2/student_stats.py
class StudentStats:
    def __init__(self, student):
        self.student = student

    def calc_average(self):
        averages = {}
        for course in self.student.get_courses():
            grades = self.student.get_grades(course)
            averages[course] = sum(grades) / len(grades)

        return averages

File: cpp/exercise2/student_stats.cpp
#include "student.cpp"
#include <iostream>

class StudentStats
{
    public:
        StudentStats(Student& student) : student(student) {}

        std::map<std::string, float> calc_average() 
        {
            std::map<std::string, float> averages;

            for (auto course : student.get_courses())
            {
                auto sum = 0;
                std::vector<int> grades = student.get_grades(course);
                for (auto grade : grades)
                {
                    sum += grade;
                }

                averages[course] = (float)sum / grades.size();
            }

            return averages;
        }

    private:
        Student& student;
};
